ludwig/models/predictor.py

```diff
@@ -124,6 +124,11 @@ def __init__(
             self.batch_evaluation = self._distributed.return_first(self.batch_evaluation)
 
     def batch_predict(self, dataset: Dataset, dataset_name: str = None, collect_logits: bool = False):
+        try:
+            self.dist_model = self._distributed.to_device(self.dist_model)
+        except AttributeError:
+            logging.info("Using DDP, skipping device assignment.")
```

Comment:
Why do we need this handler? DDP just inherits the base implementation.

Reply:
Hmm. I ran into an `AttributeError calling `distributed.to_device` in a test using `DDPStrategy`. Double-checking that now.

Reply:
I can't recreate the `AttributeError`, it may have been a separate mistake on my part while testing the fix.

Reply:
It showed up in the unit tests, but when trying to use `DistributedDataParallel` not `DDPStrategy`. It looks like the exception is raised when calling `model.to_device` inside of `BaseStrategy.to_device`.

```
ray.exceptions.RayTaskError(AttributeError): ray::_Inner.train() (pid=2974, ip=10.1.110.4, repr=TorchTrainer)
E                 File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/ray/tune/trainable/trainable.py", line 368, in train
E                   raise skipped from exception_cause(skipped)
E                 File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/ray/train/_internal/utils.py", line 54, in check_for_failure
E                   ray.get(object_ref)
E               ray.exceptions.RayTaskError(AttributeError): ray::RayTrainWorker._RayTrainWorker__execute() (pid=3026, ip=10.1.110.4, repr=<ray.train._internal.worker_group.RayTrainWorker object at 0x7ff7b49fe650>)
E                 File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/ray/train/_internal/worker_group.py", line 31, in __execute
E                   raise skipped from exception_cause(skipped)
E                 File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/ray/train/_internal/utils.py", line 129, in discard_return_wrapper
E                   train_func(*args, **kwargs)
E                 File "/home/runner/work/ludwig/ludwig/ludwig/backend/ray.py", line 498, in <lambda>
E                   lambda config: train_fn(**config),
E                 File "/home/runner/work/ludwig/ludwig/ludwig/backend/ray.py", line 215, in train_fn
E                   results = trainer.train(train_shard, val_shard, test_shard, return_state_dict=True, **kwargs)
E                 File "/home/runner/work/ludwig/ludwig/ludwig/distributed/base.py", line 155, in wrapped
E                   res = fn(*args, **kwargs)
E                 File "/home/runner/work/ludwig/ludwig/ludwig/trainers/trainer.py", line 812, in train
E                   should_break = self._train_loop(
E                 File "/home/runner/work/ludwig/ludwig/ludwig/trainers/trainer.py", line 981, in _train_loop
E                   should_break = self.run_evaluation(
E                 File "/home/runner/work/ludwig/ludwig/ludwig/trainers/trainer.py", line 572, in run_evaluation
E                   self.evaluation(test_set, TEST, progress_tracker.test_metrics, self.eval_batch_size, progress_tracker)
E                 File "/home/runner/work/ludwig/ludwig/ludwig/trainers/trainer.py", line 1074, in evaluation
E                   metrics, _ = predictor.batch_evaluation(dataset, collect_predictions=False, dataset_name=dataset_name)
E                 File "/home/runner/work/ludwig/ludwig/ludwig/models/predictor.py", line 219, in batch_evaluation
E                   self.dist_model = self._distributed.to_device(self.dist_model)
E                 File "/home/runner/work/ludwig/ludwig/ludwig/distributed/base.py", line 53, in to_device
E                   return model.to_device(device if device is not None else get_torch_device())
E                 File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1695, in __getattr__
E                   raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
E               AttributeError: 'DistributedDataParallel' object has no attribute 'to_device'
```